src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -324,25 +326,30 @@ protected override void Dispose(bool disposing)
             if (disposing)
             {
                 ProjectContextUnloaded?.Invoke(this, new ProjectContextEventArgs(this));
-
-                try
-                {
-                    foreach (var provider in SubTreeProviders)
-                    {
-                        provider.Value.DependenciesChanged -= OnDependenciesChanged;
-                    }
-                }
-                catch(ObjectDisposedException)
-                {
-                    // do nothing - we are cleaning up here and sometimes CPS throws Object disposed exception
-                }
             }
 
             base.Dispose(disposing);
         }
 
+        private Task OnUnconfiguredProjectUnloading(object sender, EventArgs args)
```

Comment:
By object you mean this current DependenciesProjectTreeView provider? I am not sure if it is necessary , since the main thing it it does handles events from subtree providers and we just unsubscribed here.